### DISCUSSION TITLE
Constructor Parameters automatically create Private Readonly Instance Fields

### DISCUSSION THREAD

**Discussion by grecosoft** [dotnet/csharplang#5090] [Language Ideas] [answered]
Since dependency-injection is now integrated into .net Core/5, having the feature, similar to Type Script, where defining a constructor's parameters prefixed with the "private" keyword automatically creates a private instance level field.  In this case, probably having the created corresponding instance field as "private readonly" might make sense.   Example before/after code:

  public class SigninService
  {

        private readonly IConfiguration _configuration;
        private readonly SignInManager<IdentityUser> _signinMgr;
        private readonly UserManager<IdentityUser> _userMgr;

        public SigninService(
            IConfiguration configuration,
            SignInManager<IdentityUser> signinMgr,
            UserManager<IdentityUser> userMgr)
        {
            _configuration = configuration;
            _signinMgr = signinMgr;
            _userMgr = userMgr;
        }
  }

    public class SigninService 
    {

        public SigninService(
            private IConfiguration _configuration,
            private SignInManager<IdentityUser> _signinMgr,
            private UserManager<IdentityUser> _userMgr)
        {
          
        }
}

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/2691 https://github.com/dotnet/csharplang/issues/4024
